db/migrate/20210610131920_restrict_sendmail_location.rb

```diff
@@ -0,0 +1,30 @@
+class FakeSetting < ApplicationRecord
+  self.table_name = 'settings'
+
+  def default=(v)
+    write_attribute :default, v.to_yaml
+  end
+
+  def value=(v)
+    v = v.to_yaml unless v.nil?
+    write_attribute :value, v
+  end
+
+  def value
+    v = read_attribute(:value)
+    YAML.load(v) unless v.nil?
+  end
+end
+
+class RestrictSendmailLocation < ActiveRecord::Migration[6.0]
+  def up
+    Setting.without_auditing do
+      existing = FakeSetting.find_by_name("sendmail_location")
+      if existing && !Setting::Email::SENDMAIL_LOCATIONS.include?(existing.value)
+        say "Sendmail location '#{existing.value}' not allowed, resetting to default"
```

Comment:
Also, suggest using settings.yaml if needed to use non-default options

Reply:
this as well

Reply:
I've added this.